### PR TITLE
LDAP security realm - iterability with pagination

### PR DESCRIPTION
**Pull request by honza889**
This is exteded #361 with pagination.
Pagination require InitialLdapContext instead of InitialDirContext, because which was added standalone SimpleLdapContextFactoryBuilder, which is not ideal - should be discussed if make SimpleDirContextFactoryBuilder less private and allow subclassing.

### CODE REVIEW

**Comment by honza889** [comment]
In the end solved simple by using only **InitialLdapContext** in SimpleDirContextFactoryBuilder.

File: src/main/java/org/wildfly/security/_private/ElytronMessages.java [hunk 1]
```diff
@@ -351,8 +351,8 @@
     @Message(id = 1077, value = "Invalid alias \"%s\" for missing mechanism database entry \"%s\"")
     void warnInvalidAliasForMissingMechanismDatabaseEntry(String value, String name);
 
-    @Message(id = 1078, value = "Ldap-backed realm failed to obtain identity from server")
-    RuntimeException ldapRealmFailedObtainIdentityFromServer(@Cause Throwable cause);
+    @Message(id = 1078, value = "Ldap-backed realm failed to obtain identity \"%s\" from server")
```
Comment (darranl) [inline src/main/java/org/wildfly/security/_private/ElytronMessages.java]: Doesn't need changing in this PR but a couple of things I think we could clean up here: -

- Some of these messages could be made more generic with the realm type being passed in.
- We should get consistent re to use single or double quotes within messages - personally I tend to stick to single as it removes the need for escaping.

**Comment by darranl** [comment]
Adding a +1 but we may still need to go further and actually expose some control of the pagination so clients can have some control if this is possible.  For now though hold off on that and that can probably be better explored when we look at console integration requirements.

**Comment by sguilhen** [comment]
I have added a +1 but it seems this PR needs some rebasing

**Comment by honza889** [comment]
Rebased.

**Pull request merged**